### PR TITLE
docs: Update README.md

### PR DESCRIPTION
**Pull request by nderjung**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

This PR updates the main README.md with the following changes:

- Correct links to documentation which have drifted over time;
- Adding information about the GitHub Action;
- Improve the "Getting started" with a 1-liner; and,
- Add a "Quick start" section which points to examples.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,71 +29,84 @@ There are many benefits in running your application as a unikernel: for more inf
...
-Simply attach a working directory on your host as a mount path volume mapped to `/workspace`, e.g.:
+KraftKit ships a container build environment which you can use instead of installing any dependencies directly on your host. It includes the `kraft` binary as well as all the additional tools and libraries for building Unikraft unikernels. Simply attach a working directory on your host as a mount path volume mapped to `/workspace`, e.g.:
 
 ```shell
 docker run -it --rm -v $(pwd):/workspace --entrypoint bash kraftkit.sh/base:latest
 ```
 
-The above command will drop you into a container shell.
-Simply type `exit` or Ctrl+D to quit.
+The above command will drop you into a container shell.  Simply type `exit` or Ctrl+D to quit.
```
Comment (craciunoiuc) [inline README.md]: Every sentence on a new line ;)

File: README.md [hunk 2]
```diff
@@ -28,71 +29,84 @@ There are many benefits in running your application as a unikernel: for more inf
...
+
+See also the [hacking documentation on how to build KraftKit from source](https://unikraft.org/docs/cli/hacking).
+
+
 ### Container build environment
 
-KraftKit ships a container build environment which you can use instead of installing any dependencies directly on your host.
-It includes the `kraft` binary as well as all the additional tools and libraries for building Unikraft unikernels.
-Simply attach a working directory on your host as a mount path volume mapped to `/workspace`, e.g.:
+KraftKit ships a container build environment which you can use instead of installing any dependencies directly on your host. It includes the `kraft` binary as well as all the additional tools and libraries for building Unikraft unikernels. Simply attach a working directory on your host as a mount path volume mapped to `/workspace`, e.g.:
```
Comment (craciunoiuc) [inline README.md]: Every sentence on a new line ;)

Applies for all other cases too

File: README.md [hunk 3]
```diff
@@ -28,71 +29,84 @@ There are many benefits in running your application as a unikernel: for more inf
...
+You can find some common project examples below:
 
-targets:
-  - name: default
-    architecture: x86_64
-    platform: qemu
-```
+| | Example |
+|-|:-|
+| ![](https://raw.githubusercontent.com/unikraft/catalog/main/icons/c.svg) | [Simple "Hello, world!" application written in C](https://github.com/unikraft/catalog/tree/main/examples/helloworld-c) |
```
Comment (craciunoiuc) [inline README.md:77]: We need to have everything here with a cool table.

We can add another column which shows if last test/build passed with a checkmark or a `Build Passing` thing

Comment (nderjung) [reply README.md:77]: We can add that column to the catalog repo :)

File: README.md (hunk 2 above)
Comment (nderjung) [reply README.md]: Ah should have been, whoops!

File: README.md [hunk 4]
```diff
@@ -51,56 +59,66 @@ docker run -it --rm -v $(pwd):/workspace --entrypoint bash kraftkit.sh/base:late
...
 
-```Makefile
-$(eval $(call addlib,apphelloworld))
 
-APPHELLOWORLD_SRCS-y += $(APPHELLOWORLD_BASE)/main.c
-```
+## Use in GitHub Actions
 
-Then it is a case of running:
+KraftKit can be used to automatically build your application into a unikernel in a GitHub Actions workflow, simply `use` `unikraft/kraftkit@staging`.
```
Comment (craciunoiuc) [inline README.md:87]:
```suggestion
KraftKit can be used to automatically build your application into a unikernel in a GitHub Actions workflow, simply use `unikraft/kraftkit@staging`.
```

File: README.md [hunk 5]
```diff
@@ -51,56 +59,66 @@ docker run -it --rm -v $(pwd):/workspace --entrypoint bash kraftkit.sh/base:late
...
+name: example
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  build:
+    steps:
+    - uses: actions/checkout@v3
```
Comment (craciunoiuc) [inline README.md]:
```suggestion
    - uses: actions/checkout@v4
```

File: README.md (hunk 4 above)
Comment (nderjung) [reply README.md:87]: `use` is intentional since it is the keyword in the action

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Pull request merged**